test/functional/common_shared_functions.go

```diff
@@ -54,31 +59,72 @@ Each resource provisioned contains an annotation with the resource ID
 This function iterates over a list of expected resource CR's
 Returns a list of resource ID's, these ID's can be used when testing AWS or GCP resources
 */
-func GetRedisInstancesIDs(ctx context.Context, client client.Client, rhmi *integreatlyv1alpha1.RHMI) ([]string, []string) {
-	var foundErrors []string
-	var foundResourceIDs []string
+func GetRedisInstanceData(ctx context.Context, client client.Client, rhmi *integreatlyv1alpha1.RHMI) (map[string]string, []error) {
+	var foundErrors []error
+	var foundData map[string]string
+	var redisVersion string
 
 	expectedRedis := getExpectedRedis(rhmi.Spec.Type, rhmi.Name)
 
 	for _, r := range expectedRedis {
```

Comment:
GitHub won't allow me to make suggestions for deleted lines but this is what I think the function should look like:
```suggestion
// GetRedisInstanceData Each resource provisioned contains an annotation with the resource ID.
// This function iterates over a list of expected Redis CRs and returns a map
// with key that represents the resource ID and value - its database version.
func GetRedisInstanceData(ctx context.Context, client client.Client, rhmi *integreatlyv1alpha1.RHMI) (map[string]string, []error) {
	var foundErrors []error
	var instanceData map[string]string
	for _, redisName := range getExpectedRedis(rhmi.Spec.Type, rhmi.Name) {
		redis := &crov1.Redis{}
		if err := client.Get(ctx, types.NamespacedName{Namespace: common.RHOAMOperatorNamespace, Name: redisName}, redis); err != nil {
			foundErrors = append(foundErrors, fmt.Errorf("failed to find %s redis cr: %w", redisName, err))
		}
		if redis.Status.Phase != croTypes.PhaseComplete {
			foundErrors = append(foundErrors, fmt.Errorf("found redis %q with phase %q but expected %q; message: %s", redisName, redis.Status.Phase, croTypes.PhaseComplete, redis.Status.Message))
		}
		resourceID, err := getCROAnnotation(redis)
		if err != nil {
			foundErrors = append(foundErrors, fmt.Errorf("redis cr %q does not contain a resource id annotation: %w", redisName, err))
		}
		instanceData[resourceID] = redis.Status.Version
	}
	return instanceData, foundErrors
}
```